### PR TITLE
Feature/hw 66553 user endpoint java create

### PR DESCRIPTION
**Pull request by Rrathinasabapath**
Please review the PR for V4 uplift of 'Create User' logic.

### CODE REVIEW

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletUser.java [hunk 1]
```diff
@@ -37,8 +46,8 @@
     private String businessName;
     private String businessOperatingName;
     private String businessRegistrationId;
-    private String businessRegistrationStateProvince;
     private String businessRegistrationCountry;
+    private String businessRegistrationStateProvince;
     private BusinessContactRole businessContactRole;
```
Comment (arao6) [inline src/main/java/com/hyperwallet/clientsdk/model/HyperwalletUser.java:51]: It looks like the Businesscontactrole field is no longer there in V4. Could you please check this - https://confluence.site1.hyperwallet.local/pages/viewpage.action?pageId=74715055

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java [hunk 2]
```diff
@@ -125,16 +133,27 @@ public void testCreateUser_withoutProgramToken() throws Exception {
         HyperwalletUser resp = client.createUser(user);
         assertThat(resp, is(equalTo(user)));
         assertThat(resp.getVerificationStatus(), is(equalTo(VerificationStatus.VERIFIED)));
+        assertThat(resp.getBusinessStakeholderVerificationStatus(), is(equalTo(BusinessStakeholderVerificationStatus.VERIFIED)));
+        assertThat(resp.getLetterOfAuthorizationStatus(), is(equalTo(LetterOfAuthorizationStatus.VERIFIED)));
+        assertThat(resp.getBusinessStakeholderVerificationStatus(), is(equalTo(BusinessStakeholderVerificationStatus.VERIFIED)));
+        assertThat(resp.getLetterOfAuthorizationStatus(), is(equalTo(LetterOfAuthorizationStatus.VERIFIED)));
```
Comment (arao6) [inline src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:139]: The above 2 lines seem to be repeated.

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletUser.java [hunk 3]
```diff
@@ -568,6 +621,29 @@ public HyperwalletUser clearGovernmentId() {
...
+    public HyperwalletUser clearGovernmentIdType() {
+        clearField("governmentIdType");
+        this.governmentIdType = null;
+        return this;
+    }
+
+
     public String getPassportId() {
         return passportId;
     }
```
Comment (arao6) [inline src/main/java/com/hyperwallet/clientsdk/model/HyperwalletUser.java:649]: The Hateoas links also need to be added to the User object.

**Review by arao6 — COMMENTED** [review]
This branch should be merged into V4

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java (hunk 2 above)
Comment (Rrathinasabapath) [reply src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:138]: Removed the duplicate lines

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletUser.java (hunk 1 above)
Comment (Rrathinasabapath) [reply src/main/java/com/hyperwallet/clientsdk/model/HyperwalletUser.java:51]: This needs clarification as there is documentation in the same page about a condition that needs businessContactRole.

**letterOfAuthorizationStatus**

Letter of authorization is required only for a business in an EEA country and only if the business contact of the business in not listed as one of the company directors

**Pull request merged**